backend/schema/metadataretry.go

```diff
@@ -0,0 +1,124 @@
+package schema
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/types/optional"
+	"google.golang.org/protobuf/proto"
+
+	schemapb "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/schema"
+)
+
+const (
+	MinBackoffLimitStr = "1s"
+	MinBackoffLimit    = 1 * time.Second
+	MaxBackoffLimitStr = "1d"
+	MaxBackoffLimit    = 24 * time.Hour
+)
+
+type MetadataRetry struct {
+	Pos Position `parser:"" protobuf:"1,optional"`
+
+	Count      *int   `parser:"'+' 'retry' (@Number Whitespace)?" protobuf:"2,optional"`
+	MinBackoff string `parser:"@(Number Ident)?" protobuf:"3"`
+	MaxBackoff string `parser:"@(Number Ident)?" protobuf:"4"`
+}
+
+var _ Metadata = (*MetadataRetry)(nil)
+
+func (*MetadataRetry) schemaMetadata()          {}
+func (m *MetadataRetry) schemaChildren() []Node { return nil }
+func (m *MetadataRetry) Position() Position     { return m.Pos }
+func (m *MetadataRetry) String() string {
+	components := []string{"+retry"}
+	if m.Count != nil {
+		components = append(components, strconv.Itoa(*m.Count))
+	}
+	components = append(components, m.MinBackoff)
+	if len(m.MaxBackoff) > 0 {
+		components = append(components, m.MaxBackoff)
+	}
+	return strings.Join(components, " ")
+}
+
+func (m *MetadataRetry) ToProto() proto.Message {
+	count := int64(*m.Count)
+	return &schemapb.MetadataRetry{
+		Pos:        posToProto(m.Pos),
+		Count:      &count,
```

Comment:
You should be able to use `proto.Int64(int64(m.Count))` here.

Reply:
Oh, looks like i made a more fundamental mistake... not handling `m.Count == 0`
Updated to
```go
var count *int64
if m.Count != nil {
  count = proto.Int64(int64(*m.Count))
}
```